homeassistant/components/roomba/vacuum.py

```diff
@@ -294,32 +269,179 @@ async def async_update(self):
+            carpet_boost = False
+            self._fan_speed = FAN_SPEED_PERFORMANCE
+        else:
+            _LOGGER.error("No such fan speed available: %s", fan_speed)
+            return
+        # The set_preference method does only accept string values
+        await self.hass.async_add_job(
+            self.vacuum.set_preference, "carpetBoost", str(carpet_boost)
+        )
+        await self.hass.async_add_job(
+            self.vacuum.set_preference, "vacHigh", str(high_perf)
+        )
+
+    async def async_update(self):
+        """Fetch state from the device."""
+        await super().async_update()
+
+        # Fan speed mode (Performance, Automatic or Eco)
+        fan_speed = None
+        carpet_boost = self.vacuum_state.get("carpetBoost")
+        high_perf = self.vacuum_state.get("vacHigh")
+        if carpet_boost is not None and high_perf is not None:
+            if carpet_boost:
+                fan_speed = FAN_SPEED_AUTOMATIC
+            elif high_perf:
+                fan_speed = FAN_SPEED_PERFORMANCE
+            else:  # carpet_boost and high_perf are False
+                fan_speed = FAN_SPEED_ECO
+        self._fan_speed = fan_speed
+
+
+class BraavaJet(IRobotBase):
+    """Braava Jet."""
+
+    def __init__(self, roomba, blid):
+        """Initialize the Roomba handler."""
+        super().__init__(roomba, blid)
+
+        # Initialize fan speed list
+        speed_list = []
+        for behavior in BRAAVA_MOP_BEHAVIORS:
+            for spray in BRAAVA_SPRAY_AMOUNT:
+                speed_list.append(f"{behavior}-{spray}")
+        self._speed_list = speed_list
+
+    @property
+    def supported_features(self):
+        """Flag vacuum cleaner robot features that are supported."""
+        return SUPPORT_BRAAVA
+
+    @property
+    def fan_speed(self):
+        """Return the fan speed of the vacuum cleaner."""
+        behavior, spray = self._fan_speed
+        return f"{behavior}-{spray}"
+
+    @property
+    def fan_speed_list(self):
+        """Get the list of available fan speed steps of the vacuum cleaner."""
+        return self._speed_list
+
+    async def async_set_fan_speed(self, fan_speed, **kwargs):
+        """Set fan speed."""
+        try:
+            split = fan_speed.split("-", 1)
+            behavior = split[0]
+            spray = int(split[1])
+            if behavior.capitalize() in BRAAVA_MOP_BEHAVIORS:
+                behavior = behavior.capitalize()
+            assert behavior in BRAAVA_MOP_BEHAVIORS
+            assert spray in BRAAVA_SPRAY_AMOUNT
+        except (IndexError, ValueError, AssertionError):
+            _LOGGER("No such fan speed available: %s", fan_speed)
+            return
+
+        overlap = 0
+        if behavior == MOP_STANDARD:
+            overlap = OVERLAP_STANDARD
+        elif behavior == MOP_DEEP:
+            overlap = OVERLAP_DEEP
+        else:
+            overlap = OVERLAP_EXTENDED
+        await self.hass.async_add_job(
```

Comment:
```suggestion
        await self.hass.async_add_executor_job(
```

async_add_job is legacy